File: server/models/thread.js
const mongoose = require("mongoose")

const ThreadSchema = new mongoose.Schema({
  username: String,
  board: { type: String, required: true },
  subject: String,
  content: { type: String, required: true },
  image: {
    url: String,
    size: Number,
    width: Number,
    height: Number,
    thumbnailWidth: Number,
    thumbnailHeight: Number
  },
  created: { type: Date, default: Date.now },
  lastBump: { type: Date, default: Date.now },
  replies: [{ type: mongoose.Schema.Types.ObjectId, ref: 'Reply' }],
  posterID: String,
  locked: { type: Boolean, default: false },
  sticky: { type: Boolean, default: false },
  upvotes: {
    count: {type: Number, default: 0},
    ids: [{type: String}]
  },
  downvotes: {
    count: {type: Number, default: 0},
    ids: [{type: String}]
  },
});

const Thread = mongoose.model('Thread', ThreadSchema);


module.exports = Thread

File: server/models/thread.go
package models;

import (
	"go.mongodb.org/mongo-driver/v2/bson"
)

type Thread struct {
	ID        bson.ObjectID      `bson:"_id,omitempty" json:"_id"`
	Username  string             `bson:"username,omitempty" json:"username,omitempty"`
	Board     string             `bson:"board" json:"board"`
	Subject   string             `bson:"subject,omitempty" json:"subject,omitempty"`
	Content   string             `bson:"content" json:"content"`
	Image     *Image             `bson:"image,omitempty" json:"image,omitempty"`
	Created   bson.DateTime      `bson:"created" json:"created"`
	LastBump  bson.DateTime      `bson:"lastBump" json:"lastBump"`
	Replies   []bson.ObjectID    `bson:"replies" json:"replies"`
	PosterID  string             `bson:"posterID,omitempty" json:"posterID,omitempty"`
	Locked    bool               `bson:"locked" json:"locked"`
	Sticky    bool               `bson:"sticky" json:"sticky"`
	Upvotes   *Votes             `bson:"upvotes" json:"upvotes"`
	Downvotes *Votes             `bson:"downvotes" json:"downvotes"`
}


